Schedule ViewModel

SidonieBouthors opened the pull request:
- [x] Show tasks on schedule
- [x] Filter shown tasks by selected events
- [x] Switch date by arrows functionality
- [x] Switch date by calendar

Polymeth commented:
The schedule displayed is not correct for me. It's displayed 2 hours before. Probably a timeslot problem

Polymeth commented on app/src/main/java/com/github/se/assocify/ui/screens/event/scheduletab/EventScheduleViewModel.kt:
```diff
@@ -0,0 +1,78 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+class EventScheduleViewModel(
```
We really need to add documentation now, even to methods with straight forward names

SidonieBouthors replied:
I added documentation :)

SidonieBouthors commented:
I fixed the offset issue in the schedule :) the same issue was occurring in the task list so I corrected it globally and added a utils class to correctly deal with offset date times and displays

Polymeth commented:
Perfect! Thanks for the work and the quick fixes :)
